### ISSUE TITLE
Simple expressions in template literals

### ISSUE THREAD

**Issue by thorn0** [prettier/prettier#3315]
**Prettier 1.8.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEBDTBeTAAwBkIAnOAW0wEsAHNAV2oBMIAbczNGrbSuDAA0mSFDRww8GIzKYAJMABG2MgF8cLejTRgaUAOYLgAMwgQ1AOkwBlOC0wx9elo1iYt-R+RjlLhAG4AHSgQIRAIOid0ZFBsDGQTbHYJcKUybDAAa0EbOkz9A2QYMkY4cLYwROTUkH0JMhgABQyDSmxqlPKQACs0AA8AIQzs3P44Yn04TtqIRhg6eYAmGe78sgbkEBUlAE9OUPC6Mn0YAHUaFhgAC2QADgAGI7IICTOMumLS7ooAR0YaBQWtg2h0kEkuuEJJQaF8ylDCuw4ABFRgQeCrcIwbBKC5XW5IJZYjI0diFADCEEo7S2UGg03CjAkABUcWhVmo1EA)

**Input:**
```jsx
const a = `Lorem ipsum dolor sit amet, consectetur ${bar} adipiscing ${foo}. Sed tincidunt diam tortor.`;

```

**Output:**
```jsx
const a = `Lorem ipsum dolor sit amet, consectetur ${bar} adipiscing ${
  foo
}. Sed tincidunt diam tortor.`;

```

**Expected behavior:**

Is the output indeed prettier than the input? To me, it looks simply strange. Why did `foo` get its own line while `bar` didn't? I'd love to have a switch to disable formatting like this. Having a line longer than `print-width` seems to be a lesser evil in this case. It should be okay though, even when the switch is on, to break lines when embedded expressions are more complex than `foo` or `data.id`.

**Comment by SimenB** [comment]
See #3299

**Comment by duailibe** [comment]
Thanks for the report! We have changed the behavior for identifiers, so it should stop breaking.

It will continue to break on member expressions (even `a.b`) until we find a good heuristic for when to break (or someone suggests one).

**Comment by thorn0** [comment]
That's great news. Thank you @duailibe. I'd vote for never breaking on member expressions if the left part is an identifier and the expression
1) uses dot (`a.b`), or
2) uses brackets with the property name being a number (`a[1]`), or
3) uses brackets with the property name being an identifier (`a[i]`).

Breaking on such simple expressions looks as unnatural as on identifiers.

**Comment by duailibe** [comment]
@thorn0 Thanks for the suggestion.

The thing about template literals is without breaking between `{` `}` we were printing:

```js
const str1 = `text text text ${foo.
  bar} text text`;

const str2 = `text text text ${foo[
  bar
]} text text`;
```

because `foo.bar` is printed regardless of being a template literal expression.

Then we could special case printing those differently when inside a template literal.

**Comment by ikatyang** [comment]
Closing as duplicate of https://github.com/prettier/prettier/issues/3280#issuecomment-345882038 and OP's `identifier` issue was fixed in #3299, let's track the `member expression` issue in #3368 to avoid confusing.
